pkg/system/phase4_configuring.go

```diff
@@ -318,6 +318,10 @@ func (r *Reconciler) SetDesiredDeploymentEndpoint() error {
 					if r.JoinSecret == nil {
 						c.Env[j].Value = r.MongoConnectionString
 					}
+				case "NOOBAA_LOG_LEVEL":
+					if r.JoinSecret == nil {
+						c.Env[j].Value = strconv.Itoa(r.DebugLevel)
```

Comment:
Again - also here we are missing the else case to unset the env if needed.

Reply:
Removed the if completely as we have a default in the yaml (0)